batch/sql/estimated-current.sql

```diff
@@ -965,6 +965,65 @@ BEGIN
   END IF;
 END $$
 
+DROP PROCEDURE IF EXISTS commit_batch_update $$
+CREATE PROCEDURE commit_batch_update(
+  IN in_batch_id BIGINT,
+  IN in_update_id INT,
+  IN in_timestamp BIGINT
+)
+BEGIN
+  DECLARE cur_update_committed BOOLEAN;
+  DECLARE expected_n_jobs INT;
+  DECLARE staging_n_jobs INT;
+
+  START TRANSACTION;
+
+  SELECT committed, n_jobs INTO cur_update_committed, expected_n_jobs
+  FROM batch_updates
+  WHERE batch_id = in_batch_id AND update_id = in_update_id
+  FOR UPDATE;
+
+  IF cur_update_committed THEN
+    COMMIT;
+    SELECT 0 as rc;
+  ELSE
+    SELECT COALESCE(SUM(n_jobs), 0) INTO staging_n_jobs
+    FROM batches_inst_coll_staging
+    WHERE batch_id = in_batch_id AND update_id = in_update_id
```

Comment:
probs want an index on batches_inst_coll_staging too

Reply:
I think we get this index for free by switching the primary key to be batch_id, update_id, inst_coll, token.